knimin/handlers/barcode_util.py

```diff
@@ -258,34 +163,135 @@ def update_ag_barcode(self, barcode):
+                        msgs=None, currentuser=self.current_user)
+            return
+
+        barcode_projects, parent_project = db.getBarcodeProjType(
+            barcode)
+        project_names = db.getProjectNames()
+
+        # barcode exists get general info
+        if barcode_details['status'] is None:
+            barcode_details['status'] = 'Received'
+        if barcode_details['biomass_remaining'] is None:
+            barcode_details['biomass_remaining'] = 'Unknown'
+        if barcode_details['sequencing_status'] is None:
+            barcode_details['sequencing_status']
+        if barcode_details['obsolete'] is None:
+            barcode_details['obsolete'] = 'N'
+        div_id = message = ""
+        ag_details = []
+        if (barcode_details['obsolete'] == "Y"):
+                # the barcode is obsolete
+                div_id = "obsolete"
+                message = "Barcode is Obsolete"
+        # get project info for div
+        if parent_project == 'American Gut':
+            div_id, message, ag_details = self.get_ag_details(barcode)
+        else:
+            div_id = "verified"
+            message = "Barcode Info is correct"
+        div_and_msg = (div_id, message, barcode)
+        self.render("barcode_util.html", div_and_msg=div_and_msg,
+                    barcode_projects=barcode_projects,
+                    parent_project=parent_project,
+                    project_names=project_names,
+                    barcode=barcode, email_type=None,
+                    barcode_info=barcode_details,
+                    proj_barcode_info=ag_details, msgs=None,
+                    currentuser=self.current_user)
+
+    @authenticated
+    def post(self):
+        barcode = self.get_argument('barcode')
+        postmark_date = self.get_argument('postmark_date', None)
+        scan_date = self.get_argument('scan_date', None)
+        biomass_remaining_value = self.get_argument('biomass_remaining_value',
+                                                    None)
+        sequencing_status = self.get_argument('sequencing_status', None)
+        obsolete_status = self.get_argument('obsolete_status', None)
+        projects = set(self.get_arguments('project'))
+        sent_date = self.get_argument('sent_date', None)
+        login_user = self.get_argument('login_user',
+                                       'American Gut participant')
+        send_mail = self.get_argument('send_mail', None)
+        login_email = self.get_argument('login_email', '')
+        other_text = self.get_argument('other_text', None)
+        email_type = self.get_argument('email_type', None)
+        sample_time = self.get_argument('sample_time', None)
+        sample_date = self.get_argument('sample_date', None)
+        # now we collect data and update based on forms
+        # first update general barcode info
+        # Set to non to make sure no conflicts with new date typing in DB
+        if not postmark_date:
```

Comment:
since the default above is `None`, failrly certain this should be `if postmark_date is not None`, but then the conditional itself doesn't make sense. This is a bit confusion